src/psyclone/psyir/transformations/increase_rank_loop_arrays_trans.py

```diff
@@ -33,26 +33,25 @@
 # -----------------------------------------------------------------------------
 # Authors: S. Siso, STFC Daresbury Lab
 
-'''This module contains the IncreaseRankLoopArrays transformation. This
-transformation increases the dimensionality of the selected arrays with
-the length of the loop iteration space, when it updates each array access
-with the variable loop, so that each access an independent element.
-Effectively it provides an alternative to array privatisation.
+'''This module contains the IncreaseRankLoopArrays transformation.'''
 
-'''
+from typing import Optional, Union
 
 from psyclone.psyGen import Transformation
 from psyclone.psyir.nodes import (
-    Loop, Reference, ArrayReference, Routine, CodeBlock)
-from psyclone.psyir.symbols import ArrayType
+    Loop, Reference, ArrayReference, Routine, CodeBlock, Range,
+    IntrinsicCall, Assignment)
+from psyclone.psyir.symbols import ArrayType, Symbol
 from psyclone.psyir.transformations.transformation_error \
     import TransformationError
 
 
 class IncreaseRankLoopArraysTrans(Transformation):
     ''' This transformation takes a loop and a list of arrays accessed inside
     the loop, and increases those arrays with an additional dimension with the
-    size of the interation space.
+    size of the interation space. Then it indexes all accesses with the loop
+    variable, so that each iteration accesses a unique location. Effectively
+    making the sub-array private for each iteration of the loop.
 
```

Comment:
Please also document the fact that it modifies all accesses outside the target Loop too. Please could you extend the example in the docstring to show this.

Reply:
Done